packages/amplify_api/lib/graphql/graphql_helpers.dart

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'package:amplify_api_plugin_interface/amplify_api_plugin_interface.dart';
+import 'package:amplify_datastore_plugin_interface/amplify_datastore_plugin_interface.dart';
+import 'package:flutter/foundation.dart';
+import '../amplify_api.dart';
+
+class ModelQueriesFactory extends ModelQueriesInterface {
+  @override
+  GraphQLRequest<T> get<T extends Model>(ModelType<T> modelType, String id) {
+    Map<String, String> variableInput = {"id": "ID"};
+
+    return GraphQLRequestFactory().buildQuery<T>(
+        modelType: modelType,
+        variableInput: variableInput,
+        id: id,
+        requestType: GraphQLRequestType.query,
+        requestOperation: GraphQLRequestOperation.get);
+  }
+}
+
+class GraphQLRequestFactory extends GraphQLRequestFactoryInterface {
+  String _getModelType(ModelFieldTypeEnum? val) {
+    switch (val) {
+      case ModelFieldTypeEnum.string:
+        return "String";
+      case ModelFieldTypeEnum.int:
+        return "int";
+      case ModelFieldTypeEnum.model:
+        return "ID";
+      default:
+        return "Error: UnknownType!";
+    }
+  }
+
+  List<String> _getFieldsFromModelType(Map<String, ModelField?>? fieldsMap) {
+    List<String> fieldsList = [];
+    if (fieldsMap != null) {
+      fieldsMap.forEach((key, value) {
```

Comment:
nit: rather than simple loop/add, could this use map function similar to https://stackoverflow.com/questions/49941361/dart-mapping-a-list-list-map? In JS I think map is better for these situations bc easier to read, dk if dart result is quite as nice.